### PR TITLE
New: Add the Otp Modal

### PR DESCRIPTION
**Pull request by shaiu**

### CODE REVIEW

File: packages/preload/src/eventsBridge.ts [hunk 1]
```diff
@@ -91,3 +91,7 @@ export async function electronGoogleOAuth2Connector(): Promise<Credentials> {
 export async function createSpreadsheet(spreadsheetId: string, credentials: Credentials): Promise<string> {
   return electron.ipcRenderer.invoke('createSpreadsheet', spreadsheetId, credentials);
 }
+
+export async function sendUserInput(input: string) {
```
[the remark this answers is not in the document]
Comment (shaiu) [reply packages/preload/src/eventsBridge.ts:95]: @baruchiro all other methods don't have error handling at all, so I don't think we should start now.

[the remark this answers is not in the document]
Comment (baruchiro) [reply packages/preload/src/eventsBridge.ts:95]: הקופיילוט די גרוע, תתעלם

**Review by baruchiro — CHANGES_REQUESTED** [review]
Please fix the issues and I will review it again.

File: packages/preload/src/eventsBridge.ts [hunk 2]
```diff
@@ -91,3 +91,7 @@ export async function electronGoogleOAuth2Connector(): Promise<Credentials> {
...
+  await electron.ipcRenderer.send('get-otp-response', input);
+}
```
Comment (baruchiro) [inline packages/preload/src/eventsBridge.ts:97]:
```suggestion
export async function sendOTPResponse(input: string) {
  await electron.ipcRenderer.send('get-otp-response', input);
}
```

File: packages/preload/src/eventsBridge.ts [hunk 3]
```diff
@@ -91,3 +91,7 @@ export async function electronGoogleOAuth2Connector(): Promise<Credentials> {
 export async function createSpreadsheet(spreadsheetId: string, credentials: Credentials): Promise<string> {
   return electron.ipcRenderer.invoke('createSpreadsheet', spreadsheetId, credentials);
 }
+
+export async function sendUserInput(input: string) {
+  await electron.ipcRenderer.send('get-otp-response', input);
```
Comment (baruchiro) [inline packages/preload/src/eventsBridge.ts:96]: Why `send` and not `invoke`?

File: packages/renderer/src/components/GetOtp.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import { useConfigStore } from '/@/store/ConfigStore';
+import { sendUserInput } from '#preload';
+import { ModalStatus } from '/@/types';
+
+const GetOtp = () => {
+  const configStore = useConfigStore();
+  const [modalStatus, setModalStatus] = useState<ModalStatus>(ModalStatus.HIDDEN);
+  const [inputText, setInputText] = useState('');
+
+  const closeModal = () => setModalStatus(ModalStatus.HIDDEN);
```
Comment (baruchiro) [inline packages/renderer/src/components/GetOtp.tsx:14]: The `ModalStatus` is to reuse the main modal (`Body.tsx`) for different usages.

Here you create a new Modal, I think you can just use a boolean state to decide if the Modal is open or not.

File: packages/preload/src/eventsBridge.ts (hunk 3 above)
Comment (shaiu) [reply packages/preload/src/eventsBridge.ts:96]: I'm still learning. What's the difference?

File: packages/renderer/src/components/GetOtp.tsx (hunk 4 above)
Comment (shaiu) [reply packages/renderer/src/components/GetOtp.tsx]: Done. changed to boolean

File: packages/preload/src/eventsBridge.ts (hunk 3 above)
Comment (baruchiro) [reply packages/preload/src/eventsBridge.ts]: I'm not sure there is a difference,

https://stackoverflow.com/a/59889863/839513

But all other calls are using `invoke`, so you should as well.

File: packages/renderer/src/components/GetOtp.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import { observer } from 'mobx-react-lite';
+import { useConfigStore } from '/@/store/ConfigStore';
+import { sendOTPResponse } from '#preload';
+
+const GetOtp = () => {
+  const configStore = useConfigStore();
+  const [modalStatus, setModalStatus] = useState<boolean>(false);
+  const [inputText, setInputText] = useState('');
+
+  const closeModal = () => setModalStatus(false);
```
Comment (baruchiro) [inline packages/renderer/src/components/GetOtp.tsx:13]: OK, you allow the user to close the modal, but the backend process will still wait for the OTP.

If the user closed the modal, I assume you should `sendOTPResponse(undefined)` or something to fail the process.

File: packages/renderer/src/components/GetOtp.module.css
Comment (baruchiro) [inline packages/renderer/src/components/GetOtp.module.css:1]: I think all the styles in this document are not used, and the only used style `modalHeader` is not declared here.

File: packages/preload/src/eventsBridge.ts (hunk 3 above)
Comment (shaiu) [reply packages/preload/src/eventsBridge.ts]: Done

File: packages/renderer/src/components/GetOtp.tsx (hunk 5 above)
Comment (shaiu) [reply packages/renderer/src/components/GetOtp.tsx]: Done

File: packages/renderer/src/components/GetOtp.module.css
Comment (shaiu) [reply packages/renderer/src/components/GetOtp.module.css:1]: Right. Removed css file

**Review by baruchiro — APPROVED** [review]

**Pull request merged**